mne/io/edf/edf.py

```diff
@@ -201,12 +202,14 @@ def _read_segment(self, start=0, stop=None, sel=None, verbose=None,
             buffer_size = blockstop - blockstart
             pointer = blockstart * n_chan * data_size
             fid.seek(data_offset + pointer)
-            datas = np.zeros((n_chan, buffer_size), dtype=float)
+            datas = np.empty((n_chan, buffer_size), dtype=float)
             blocks = int(ceil(float(buffer_size) / sfreq))
             if 'n_samps' in self._edf_info:
                 n_samps = self._edf_info['n_samps']
             # bdf data: 24bit data
-            if self._edf_info['subtype'] == '24BIT':
+            if subtype == '24BIT' or subtype == 'bdf':
```

Comment:
could also be `if subtype in ('24BIT', 'bdf'):` if you feel like using fewer chars :)